src/platform/forms-system/src/js/containers/FormPage.jsx

```diff
@@ -90,6 +90,15 @@ class FormPage extends React.Component {
 
     const path = getNextPagePath(route.pageList, form.data, location.pathname);
 
+    if (typeof route.pageConfig.onNavForward === 'function') {
+      route.pageConfig.onNavForward({
+        formData,
+        goPath: customPath => this.props.router.push(customPath),
+        goNextPath: () => this.props.router.push(path),
+      });
```

Comment:
If this were used for non-CustomPages then I think we'd need to pass in a page index, in case we're navigating inside an array path

Reply:
Good catch.  I'll be continuing to iterate this as I build out the consumer implementation for a pattern, so I will add this in follow up PRs